cpp/src/parquet/column_reader.h

```diff
@@ -302,7 +300,284 @@ class TypedColumnReader : public ColumnReader {
                                           int32_t* dict_len) = 0;
 };
 
+// Represent a range to read. The range is inclusive on both ends.
+struct IntervalRange {
+  static IntervalRange Intersection(const IntervalRange& left,
+                                    const IntervalRange& right) {
+    if (left.start <= right.start) {
+      if (left.end >= right.start) {
+        return {right.start, std::min(left.end, right.end)};
+      }
+    } else if (right.end >= left.start) {
+      return {left.start, std::min(left.end, right.end)};
+    }
+    return {-1, -1};  // Return a default Range object if no intersection range found
+  }
+
+  IntervalRange(const int64_t start_, const int64_t end_) : start(start_), end(end_) {
+    if (start > end) {
+      throw ParquetException("Invalid range with start: " + std::to_string(start) +
+                             " and end: " + std::to_string(end));
+    }
+  }
+
+  size_t Count() const { return end - start + 1; }
```

Comment:
`IntervalRange({-1, -1}).Count() == 1` ? Would this be weird?

Reply:
added a check of validness